transactions/demo-typescript/src/lib/query.ts

```diff
@@ -0,0 +1,444 @@
+// deno-lint-ignore-file no-explicit-any -- mismatches between Ogmios' types vs the actual data
+/**
+ * Functionality to isolate how we query the Cardano node
+ */
+import {
+  createInteractionContext,
+  createLedgerStateQueryClient,
+  createTransactionSubmissionClient,
+  InteractionContext,
+} from "@cardano-ogmios/client";
+import * as ogmios from "@cardano-ogmios/schema";
+import * as csl from "@emurgo/cardano-serialization-lib-nodejs";
+import { TxBuilderConfig, TxExUnits } from "./tx-builder.js";
+import { ogmiosUtxoToCslUtxo } from "./utils.js";
+
+/**
+ * {@link Query} describes functionality for interaction with a Cardano
+ * node to make this demo project work
+ */
+export interface Query {
+  queryAddressUtxos(addr: csl.Address): Promise<csl.TransactionUnspentOutputs>;
+  queryUtxo(
+    transactionInput: csl.TransactionInput,
+  ): Promise<csl.TransactionUnspentOutput | undefined>;
+
+  submitTx(transaction: csl.Transaction): Promise<csl.TransactionHash>;
+  awaitTx(
+    transactionHash: csl.TransactionHash,
+    pollDelayMs?: number,
+    maxCount?: number,
+  ): Promise<void>;
+  /**
+   * Queries the execution units of the transaction.
+   *
+   * @param excessBudgetFactor - what to multiply the budgets by after
+   * evaluating it. See the warning {@link https://ogmios.dev/mini-protocols/local-tx-submission/#evaluating-transactions}
+   */
+  evaluateTx(
+    transaction: csl.Transaction,
+    excessBudgetFactor?: number,
+  ): Promise<TxExUnits>;
```

Comment:
You'd probably want to have a Submit interface or smtn

Reply:
I see, I split `Query` into two interfaces:
- `Query` for `Query`ing things
- `Submit` for `Submit` transactions.
Note that Ogmios allows one to do both, and hence implements both interfaces